train.py

```diff
@@ -1 +1,241 @@
+              validation_data=(x_validate, y_validate), callbacks=[checkpointer, tensorboard])
+
+    return model
+
+
+def test(epochs, saving_period, x_test, y_test, load_dir, model):
+    """
+    This module tests model after each training epoch on testing dataset and shows score.
+
+    :param epochs:
+        Number of training epochs.
+
+    :param saving_period:
+        How many epochs of training must pass before saving model.
+
+    :param x_test:
+        Input testing data.
+
+    :param y_test:
+        Output testing data.
+
+    :param load_dir:
+        Path for loading model.
+
+    :param model
+        Tested  neural network model.
+    """
+    for i in range(int(epochs/saving_period)):
+        # Downloading weights
+        try:
+            model = load_model(load_dir.format(i+1))
+        except:
+            print("Unable to load model")
+        # Checking results with testing dataset
+        score_test = model.evaluate(x_test, y_test, verbose=1)
+        print("Score after {}. epoch on testing dataset: ".format(i+1))
+        print("Loss:")
+        print(score_test[0])
+        print("Accuracy:")
+        print(score_test[1])
+
+
+def parse_args():
+    """
+    This module parses command line arguments.
+
+    :return:
+        Parser containing command line data.
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--dataset', '-d', metavar="csv file containing dataset",
+                        help='input file containing dataset required in training', default='smile_warrior_dataset.csv')
+
+    parser.add_argument('--batch_size', '-b', metavar="number of images in one batch",
+                        help='number of images in one batch', type=int, default=32)
+
+    parser.add_argument('--epochs', '-e', metavar="number of training epochs",
+                        help='number of training epochs', type=int, default=10)
+
+    parser.add_argument('--saving_period', '-s', metavar="how many epochs must pass before saving",
+                        help='how many epochs must pass before saving', type=int, default=1)
+
+    parser.add_argument('--learning_rate', '-lr', metavar="specifies learning rate",
+                        help='specifies learning rate', type=float, default=0.0001)
+
+    parser.add_argument('--models_dir', '-md', metavar="specifies saving directory for models",
+                        help='specifies saving directory for models', default='model.{epoch:02d}.hdf5')
+
+    parser.add_argument('--logs_dir', '-ld', metavar="specifies saving directory for logs",
+                        help='specifies saving directory for logs', default='logs')
+
+    parser.add_argument('--model_load', '-ml', metavar="specifies path for loading models",
+                        help='specifies path for loading models', default="model.0{}.hdf5")
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    # Network parameters
+    batch_size = args.batch_size
```

Comment:
I would not unpack these, but I leave it to your choice.

Reply:
I think I'll leave it as it is.